src/python/bot/tasks/train_gradientfuzz_task.py

```diff
@@ -13,9 +13,308 @@
+
+  Returns:
+    (new_process.ProcessResult): Result of `run_and_wait()`.
+    (str): Dataset name (results stored under
+        GRADIENTFUZZ_SCRIPTS_DIR/data/[dataset_name]).
+  """
+  script_path = get_script_path(run_constants.GENERATE_DATA_SCRIPT)
+  dataset_name = os.path.basename(corpus_directory)
+  args_list = [
+      script_path,
+      run_constants.FUZZ_TARGET_BINARY_FLAG,
+      fuzzer_binary_path,
+      run_constants.INPUT_DIR_FLAG,
+      corpus_directory,
+      run_constants.DATASET_NAME_FLAG,
+      dataset_name,
+      run_constants.MEDIAN_MULT_FLAG,
+      run_constants.DEFAULT_MEDIAN_MULT_CUTOFF,
+  ]
+
+  script_environment = os.environ.copy()
+
+  logs.log('Launching input gen with args: "{}".'.format(str(args_list)))
+
+  # Run process in GradientFuzz directory.
+  data_gen_proc = new_process.ProcessRunner(sys.executable)
+
+  return data_gen_proc.run_and_wait(
+      additional_args=args_list,
+      cwd=GRADIENTFUZZ_SCRIPTS_DIR,
+      env=script_environment,
+      timeout=run_constants.DATA_GEN_TIMEOUT), dataset_name
+
+
+def train_gradientfuzz(fuzzer_name, dataset_name, num_inputs, testing):
+  """Train GradientFuzz model.
+
+  Args:
+    fuzzer_name (str): Prefix to --run-name flag.
+    dataset_name (str): Inputs/labels stored under
+        GRADIENTFUZZ_SCRIPTS_DIR/data/[dataset_name].
+    num_inputs (int): Number of input files (for val split/batch size).
+    testing (bool): Whether or not we use testing number of epochs.
+
+  Returns:
+    (new_process.ProcessResult): Result of `run_and_wait()`.
+    (str): Run name (results stored under
+        GRADIENTFUZZ_SCRIPTS_DIR/models/[architecture]/[run_name]).
+  """
+  if num_inputs < run_constants.MIN_NUM_INPUTS:
+    return new_process.ProcessResult(
+        return_code=run_constants.ExitCode.CORPUS_TOO_SMALL), None
+
+  batch_size = str(4) if testing else str(min(32, int(num_inputs * 0.4)))
+  val_batch_size = str(4) if testing else str(min(32, int(num_inputs * 0.1)))
+  num_epochs = str(run_constants.NUM_TEST_EPOCHS
+                   if testing else run_constants.NUM_EPOCHS)
+
+  script_path = get_script_path(run_constants.TRAIN_MODEL_SCRIPT)
+  run_name = fuzzer_name + run_constants.RUN_NAME_SUFFIX
+  args_list = [
+      script_path, run_constants.RUN_NAME_FLAG, run_name,
+      run_constants.DATASET_NAME_FLAG, dataset_name, run_constants.EPOCHS_FLAG,
+      num_epochs, run_constants.BATCH_SIZE_FLAG, batch_size,
+      run_constants.VAL_BATCH_SIZE_FLAG, val_batch_size,
+      run_constants.ARCHITECTURE_FLAG, constants.NEUZZ_ONE_HIDDEN_LAYER_MODEL
+  ]
+
+  script_environment = os.environ.copy()
+
+  logs.log('Launching training with the following arguments: "{}".'.format(
+      str(args_list)))
+
+  # Run process in gradientfuzz directory.
+  gradientfuzz_trainer = new_process.ProcessRunner(sys.executable)
+
+  return gradientfuzz_trainer.run_and_wait(
+      args_list,
+      cwd=GRADIENTFUZZ_SCRIPTS_DIR,
+      env=script_environment,
```

Comment:
Since this isn't being modified, copying/passing the env isn't necessary. 

Same also on line 185.

Reply:
Okay great! I wasn't 100% clear on the idea of `os.environ.copy()` there.